tools/build/AppImage/build-with-conda.sh

```diff
@@ -0,0 +1,209 @@
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+# adapted from https://github.com/FreeCAD/FreeCAD-Bundle/blob/master/conda/linux/create_bundle.sh
+
+# This script produces a FreeCAD AppImage locally using conda repositories. It
+# is an all-in-one script that fetches FreeCAD and all dependencies from conda
+# repo (it also extracts a couple of XDG files from the FreeCAD source). Everything
+# comes from conda repositories, therefore, you need to make sure they contain the
+# FreeCAD package you want to build. An easy way to check is by searching for
+# available FreeCAD packages:
+#
+#     conda search "freecad=0.21.0" -c adrianinsaval/label/dev -c freecad -c conda-forge
+#
+
+
+# To use this script, you will need:
+#
+#     1) Mambaforge from https://github.com/conda-forge/miniforge#mambaforge
+#     if you choose to not install the conda environment when installing Mambaforge,
+#     you will need to do it before running this script, by running the following
+#     in your terminal. This will make the "conda" and "mamba" commands available:
+#     eval "$(/home/yorik/Mambaforge/bin/conda shell.zsh hook)"
+#     (replace the Mambaforge path by yours and zsh by your shell name)
+#
+#     2) appimagetool from https://appimage.github.io/appimagetool/
+#     place it in your exec path so it can be found by this script
+#
+#     3) a gpg key to sign the package
+#
+#     4) verify or change the config values here below
+#
+#     5) If you are on Fedora, there are lines to uncomment below in the script
+#
+# When done, just run this script and you should get an appimage with the
+# corresponding sha hash in the current folder.
+#
+# To cleanup after build: Just delete the "AppDir" folder (not done automatically since
+# next builds can reuse the downloaded packages).
+
+# config
+
+# make sure you have a gpg key for this email
+gpg_key="yorik@freecad.org"
+# the FreeCAD version we're looking for
+target_version="0.21.0"
+# make sure target_python matches the one FreeCAD is built with! Check with
+# conda search "freecad=0.21.0" -c adrianinsaval/label/dev -c freecad -c conda-forge
+target_python="3.10"
```

Comment:
this could be made into optional arguments (that default to these values) so the user doesn't actually need to edit the script to use it.

Reply:
the conda channel to be used could be a good thing to put as an argument for the script, this way people could create their own conda channel and generate appimages for their forks (e.g. realthunder, althought he seems to already have a good system in place)